KPCTabsControl/TabsControl.swift

```diff
@@ -461,12 +464,24 @@ public class TabsControl: NSControl, TabEditingDelegate {
      - parameter index: An integer indicating the index of the item to be selected.
      */
     public func selectItemAtIndex(index: Int) {
-        let buttons = self.tabButtons()
-        if buttons.count > index {
-            self.selectTab(buttons[index])
+        guard let button = self.tabButtons()[safe: index] else { return }
+
+        self.selectTab(button)
+    }
+
+    private func updateButtonStatesForSelection() {
+        for button in self.tabButtons() {
+            if let equatableSelectedItem = selectedItem as? String,
```

Comment:
I thought instance equality would work. But I see that it can be misleading if it is a kind of easy-to-produce volatile thing such as a literal string. Working with Indexes sounds good to me too (b).